src/lib/Parser.hs

```diff
@@ -162,7 +162,10 @@ exprAsModule :: UExpr -> (Name, UModule)
 exprAsModule e = (asGlobal v, UModule (toNest [d]))
   where
     v = mkName "_ans_"
-    d = ULet PlainLet (WithSrc (srcPos e) (UPatBinder (Bind (v:>()))), Nothing) e
+    d = sourceLet (WithSrc (srcPos e) (UPatBinder (Bind (v:>()))), Nothing) e
+
+sourceLet :: UPatAnn -> UExpr -> UDecl
+sourceLet = ULet (PlainLet SomeEffects)
```

Comment:
Will this `SomeEffects` annotation get later refined to `NoEffects` if the expr doesn't have them?